prototypes/technology/trainassembly-technology.lua

```diff
@@ -0,0 +1,151 @@
+require "util"
+
+local trainTech = util.table.deepcopy(data.raw["technology"]["railway"])
+
+  trainTech.name = "trainassembly-trainTechnology"
+  trainTech.effects = {}
+  trainTech.localised_name = {"technology-name.trainTech"}
+  trainTech.localised_description = {"technology-description.trainTech"}
+
+  data:extend({
+    trainTech,
+  })
+
+for _, trainTechn in pairs ({
+  "locomotive",
+  "cargo-wagon",
+}) do
```

Comment:
you only unlock the locomotive parts, remember you'll also have to unlock the fluids
PS: this also have to happen for arty and fluid wagon